### PR TITLE
Added changes for new cluster objects attributes

### PR DESCRIPTION
**Pull request by shtripat**
tendrl-bug-id: Tendrl/commons#797
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by shtripat** [comment]
@r0h4n @nthomas-redhat @GowthamShanmugam plz review

**Comment by shtripat** [comment]
@r0h4n the commons changes for cluster objects are done as part of https://github.com/Tendrl/commons/pull/798/

File: tendrl/gluster_integration/manager/__init__.py [hunk 1]
```diff
@@ -78,6 +80,39 @@ def shutdown(signum, frame):
             {"message": "Signal handler: stopping"}
         )
         complete.set()
```
Comment (r0h4n) [inline tendrl/gluster_integration/manager/__init__.py]: complete.set() should be at the end of this function

File: tendrl/gluster_integration/manager/__init__.py [hunk 2]
```diff
@@ -78,6 +80,39 @@ def shutdown(signum, frame):
...
+        try:
+            gl_srvr_list = NS._int.client.read(
+                "/indexes/tags/gluster/server"
+            ).value
+            gl_srvr_list = json.loads(gl_srvr_list)
+            if NS.node_context.node_id in gl_srvr_list:
+                gl_srvr_list.remove(NS.node_context.node_id)
+            NS._int.client.write(
+                "/indexes/tags/gluster/server",
+                gl_srvr_list
```
Comment (r0h4n) [inline tendrl/gluster_integration/manager/__init__.py]: youll need to convert "gl_srvr_list" back to json

File: tendrl/gluster_integration/manager/__init__.py [hunk 3]
```diff
@@ -78,6 +80,39 @@ def shutdown(signum, frame):
...
+            )
+            NS._int.client.delete(
```
Comment (r0h4n) [inline tendrl/gluster_integration/manager/__init__.py]: check if this service actually is the provisioner and then delete the tag

File: tendrl/gluster_integration/manager/__init__.py (hunk 1 above)
Comment (shtripat) [reply tendrl/gluster_integration/manager/__init__.py]: done

File: tendrl/gluster_integration/manager/__init__.py (hunk 2 above)
Comment (shtripat) [reply tendrl/gluster_integration/manager/__init__.py]: its a python list only at this moment

File: tendrl/gluster_integration/manager/__init__.py (hunk 3 above)
Comment (shtripat) [reply tendrl/gluster_integration/manager/__init__.py]: ok

File: tendrl/gluster_integration/manager/__init__.py [hunk 4]
```diff
@@ -77,6 +79,42 @@ def shutdown(signum, frame):
             NS.publisher_id,
             {"message": "Signal handler: stopping"}
         )
+        # Remove the node's name from gluster server tag
+        try:
+            gl_srvr_list = NS._int.client.read(
+                "/indexes/tags/gluster/server"
```
Comment (r0h4n) [inline tendrl/gluster_integration/manager/__init__.py]: Also handle tag `tendrl/integration/gluster`
https://github.com/Tendrl/commons/blob/master/tendrl/commons/objects/definition/master.yaml#L12

File: tendrl/gluster_integration/manager/__init__.py (hunk 2 above)
Comment (r0h4n) [reply tendrl/gluster_integration/manager/__init__.py]: You need to convert "gl_srvr_list" to json before writing to etcd.

https://github.com/Tendrl/node-agent/blob/master/tendrl/node_agent/node_sync/services_and_index_sync.py#L146

File: tendrl/gluster_integration/manager/__init__.py [hunk 5]
```diff
@@ -77,6 +79,42 @@ def shutdown(signum, frame):
...
+            ).value
+            gl_srvr_list = json.loads(gl_srvr_list)
+            if NS.node_context.node_id in gl_srvr_list:
+                gl_srvr_list.remove(NS.node_context.node_id)
+            NS._int.client.write(
```
Comment (r0h4n) [inline tendrl/gluster_integration/manager/__init__.py]: use  commons etcd utils for writes

File: tendrl/gluster_integration/manager/__init__.py [hunk 6]
```diff
@@ -77,6 +79,42 @@ def shutdown(signum, frame):
             NS.publisher_id,
             {"message": "Signal handler: stopping"}
         )
+        # Remove the node's name from gluster server tag
+        try:
+            gl_srvr_list = NS._int.client.read(
```
Comment (r0h4n) [inline tendrl/gluster_integration/manager/__init__.py]: use commons etcd utils to read

File: tendrl/gluster_integration/manager/__init__.py (hunk 2 above)
Comment (shtripat) [reply tendrl/gluster_integration/manager/__init__.py]: ok

File: tendrl/gluster_integration/manager/__init__.py (hunk 4 above)
Comment (shtripat) [reply tendrl/gluster_integration/manager/__init__.py]: ok

File: tendrl/gluster_integration/manager/__init__.py (hunk 5 above)
Comment (shtripat) [reply tendrl/gluster_integration/manager/__init__.py]: ok

File: tendrl/gluster_integration/manager/__init__.py (hunk 6 above)
Comment (shtripat) [reply tendrl/gluster_integration/manager/__init__.py]: ok

**Pull request merged**